File: ios/AppfastflyClipboard.m
#import "AppfastflyClipboard.h"

@implementation AppfastflyClipboard

- (NSString *)getTokenWithPrefix:(NSString *)prefix {
  NSString *content = [UIPasteboard generalPasteboard].string;
  if (!content) return nil;
  if (![content hasPrefix:prefix]) return nil;
  return [content substringFromIndex:prefix.length];
}

- (void)clear {
  [UIPasteboard generalPasteboard].string = @"";
}

@end

File: ios/AppfastflyClipboard.swift
import UIKit

@objc(AppfastflyClipboard)
public class AppfastflyClipboard: NSObject {

  @objc public func getToken(prefix: String) -> String? {
    guard let content = UIPasteboard.general.string else { return nil }
    guard content.hasPrefix(prefix) else { return nil }
    return String(content.dropFirst(prefix.count))
  }

  @objc public func clear() {
    UIPasteboard.general.string = ""
  }
}
